### ISSUE TITLE
bug: [Trainer controller] In trainer controller test cases, conditionals should use “is” instead of ==

### ISSUE THREAD

**Issue by seshapad** [foundation-model-stack/fms-hf-tuning#186]
## Describe the bug

Trainer controller test cases are using the following conditional statement in assert:
```
    assert control.should_training_stop == True
```
Above statement needs to be changed to:
```
    assert control.should_training_stop is True
```

## Platform

Please provide details about the environment you are using, including the following:

- Interpreter version:
- Library version: [X]

## Sample Code

NA

## Expected behavior

NA

## Observed behavior
NA

## Additional context

NA

**Comment by anhuong** [comment]
@seshapad can you provide more details and an example? Is this just for comparisons to None?

**Comment by seshapad** [comment]
> @seshapad can you provide more details and an example? Is this just for comparisons to None?

@anhuong I have added details and an example.

**Comment by seshapad** [comment]
Addressed in PR [#169](https://github.com/foundation-model-stack/fms-hf-tuning/pull/169).
